pythonAPIClient/client.py

```diff
@@ -420,43 +421,42 @@ def add_camera(self, camera_type, is_active_image, is_active_video, snapshot_url
                     The new camera ID for the successfully updated camera.
         """
 
-        local_params = dict(locals())
+        required_args = ('type', 'is_active_image', 'is_active_video')
+
+        self._check_args(kwargs=kwargs, required_args=required_args)
 
         if self.token is None:
             self.request_token()
 
         url = Client.base_URL + 'cameras/create'
 
-        local_params['type'] = local_params.pop('camera_type')
-        del local_params['self']
-
-        if camera_type == 'ip':
-            if ip is not None:
-                local_params['retrieval'] = {
-                    'ip': local_params.pop('ip'),
-                    'port': local_params.pop('port'),
-                    'brand': local_params.pop('brand'),
-                    'model': local_params.pop('model'),
-                    'image_path': local_params.pop('image_path'),
-                    'video_path': local_params.pop('video_path')
+        if kwargs.get('type') == 'ip':
+            if kwargs.get('ip') is not None:
+                kwargs['retrieval'] = {
+                    'ip': kwargs.pop('ip', None),
+                    'port': kwargs.pop('port', None),
+                    'brand': kwargs.pop('brand', None),
+                    'model': kwargs.pop('model', None),
+                    'image_path': kwargs.pop('image_path', None),
+                    'video_path': kwargs.pop('video_path', None)
                 }
-                local_params['retrieval'] = json.dumps(local_params['retrieval'])
-        elif camera_type == 'non-ip':
-            if snapshot_url is not None:
-                local_params['retrieval'] = {
-                    'snapshot_url': local_params.pop('snapshot_url')
+                kwargs['retrieval'] = json.dumps(kwargs['retrieval'])
+        elif kwargs.get('type') == 'non-ip':
+            if kwargs.get('snapshot_url') is not None:
+                kwargs['retrieval'] = {
+                    'snapshot_url': kwargs.pop('snapshot_url', None)
```

Comment:
we can simplify retrieval of by `kwargs['retrieval'] = { 'snapshot_url' : kwargs.pop('snapshot_url', None)` without first checking if it is there. if the any of the fields are not specified, they will be None and error will be returned by the api.
same for ip and stream camera.